integration/test/metrics_number_dimension/metrics_number_dimension_test.go

```diff
@@ -52,36 +47,45 @@ func TestNumberMetricDimension(t *testing.T) {
 	parameters := []input{
 		{
 			resourcePath:         "resources/10_dimension",
-			findTarget:           false,
+			failToStart:          false,
 			numberDimensionsInCW: 10,
 			metricName:           "mem_used_percent",
 		},
-		// @TODO add when https://github.com/aws/amazon-cloudwatch-agent/pull/361 is merged
-		// {resourcePath: "resources/30_dimension", findTarget: false, numberDimensionsInCW: 30, metricName: "mem_used_percent",},
-		// {resourcePath: "resources/35_dimension", findTarget: true, numberDimensionsInCW: 30, metricName: "mem_used_percent",},
+		{
+			resourcePath:         "resources/30_dimension",
+			failToStart:          false,
+			numberDimensionsInCW: 30,
+			metricName:           "mem_used_percent",
+		},
+		{
+			resourcePath:         "resources/35_dimension",
+			failToStart:          true,
+			numberDimensionsInCW: -1,
+			metricName:           "mem_used_percent",
+		},
 	}
 
 	for _, parameter := range parameters {
 		//before test run
-		log.Printf("resource file location %s find target %t input number dimension %d metric name %s",
-			parameter.resourcePath, parameter.findTarget, parameter.numberDimensionsInCW, parameter.metricName)
+		log.Printf("resource file location %s fail to start %t input number dimension %d metric name %s",
+			parameter.resourcePath, parameter.failToStart, parameter.numberDimensionsInCW, parameter.metricName)
 
-		target := fmt.Sprintf(targetString, maxDimension, parameter.numberDimensionsInCW)
-
-		t.Run(fmt.Sprintf("resource file location %s find target %t", parameter.resourcePath, parameter.findTarget), func(t *testing.T) {
+		t.Run(fmt.Sprintf("resource file location %s find target %t", parameter.resourcePath, parameter.failToStart), func(t *testing.T) {
 			test.CopyFile(parameter.resourcePath+configJSON, configOutputPath)
-			test.StartAgent(configOutputPath)
+			err := test.StartAgent(configOutputPath, false)
+
+			// for append dimension we auto fail over 30 for custom metrics (statsd we collect remove dimension and continue)
+			if parameter.failToStart && err == nil {
+				t.Fatalf("Agent should not have started for append %v dimension", parameter.resourcePath)
```

Comment:
Should this be 
```suggestion
				t.Fatalf("Agent should not have started for append %v dimension", parameter.numberDimensionsInCW)
```
Maybe I just don't understand the test table params anymore

Reply:
I will change this after the integration tests are done running so they will not have commit hash issues on pull.